cs1_lab01.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+number: 1
+title: C programming 1
+author:
+course: Computer Science 1
+material: Instruction 1
+---
+
+## 1.
+Create and execute the following program:
+```c++
+#include <stdio.h>          // header for standard input/output
+
+int main()                  // definition of the main function
+{                           // main function starting bracket
+  printf("Hello!\n");
+  return 0;
+}                           // main function ending bracket
+```
+Declare two integer variables (a and b) and print results of: sum, subtraction, multiplication and division.
+The results should be printed in a single line.
+
+## 2.
+Write a program which reads from a keyboard two numbers of double type and prints results of the following operations: multiplication, division, square root, cube root, fourth power, sinus and cosinus.
+Print information on what kind of input is expected before the program starts waiting for the input:
+```c++
+printf("Input the first number:\n");
+scanf("%lf", &a);
+```
+For example, the printout should be formated as follows:
+```
+First number    : 12.4
+Second number   : 32.22
+Multiplication  : 399.5
+Division        : 3.85e-001
+Square roots    : |3.52| |5.68|
+Cube roots      : 2.314589 3.182061
+Fourth powers   : 23642 1077711
+```
+Hint: use tabulator.
+
+## 3.
+Write a program which reads from a keyboard coordinates of three arbitrary points in 2D.
+Calculate distances between those points.
+Calculate also an area and a perimeter of a triangle defined by these points.
+Use Heron formula $P = \sqrt{s(s−a)(s−b)(s−c)}$, where $s= \frac{1}{2} (a + b + c )$, and $a$, $b$, $c$ are the lengths of the triangle sides.
+Format a printout of the results in a clear and readable way.
+
+## 4.
+Modify the program from the previous point by adding code calculating angles of the triangle according to the following formulas:
+$$
+\tan\left(\frac{\gamma}{2} \right) = \sqrt{\frac{(p-a)(p-b)}{p(p-c)}}, \quad
+\tan\left(\frac{\beta}{2} \right) = \sqrt{\frac{(p-a)(p-c)}{p(p-b)}},
+\quad
+\tan\left(\frac{\alpha}{2} \right) = \sqrt{\frac{(p-b)(p-c)}{p(p-a)}},
+$$
+Print the angles in degrees with precision to three decimal places.
+Check also if the calculated angles fulfill the condition: $\pi = \alpha + \beta + \gamma$.
+The number $\pi$ should be defined using a preprocessor command:
+```c++
+#define PI (4.*atan(1))
```

Comment:
Can we please stop teaching pi this way? Maybe use [`M_PI`](https://stackoverflow.com/questions/1727881/how-to-use-the-pi-constant-in-c) or wrap [`std::numbers::pi_v<double>`](https://en.cppreference.com/w/cpp/numeric/constants) in a macro if C++20 is available (since we're using a C++ compiler anyway)